src/modules/Client/Api/Guest.php

```diff
@@ -150,78 +151,99 @@ public function login($data)
     public function reset_password($data)
     {
         $this->di['events_manager']->fire(['event' => 'onBeforePasswordResetClient']);
-        $required = [
-            'email' => 'Email required',
-        ];
-        $this->di['validator']->checkRequiredParamsForArray($required, $data);
+
+        // Validate required parameters
+        $this->di['validator']->checkRequiredParamsForArray(['email' => 'Email required'], $data);
+
+        // Sanitize email
         $data['email'] = $this->di['tools']->validateAndSanitizeEmail($data['email']);
 
         $this->di['events_manager']->fire(['event' => 'onBeforeGuestPasswordResetRequest', 'params' => $data]);
 
+        // Fetch the client by email
         $c = $this->di['db']->findOne('Client', 'email = ?', [$data['email']]);
         if (!$c instanceof \Model_Client) {
             return true;
         }
 
-        $hash = hash('sha256', time() . random_bytes(13));
+        // Check if a password reset request exists
+        $reset = $this->di['db']->findOne('ClientPasswordReset', 'client_id = ?', [$c->id]);
 
-        $reset = $this->di['db']->dispense('ClientPasswordReset');
-        $reset->client_id = $c->id;
-        $reset->ip = $this->ip;
-        $reset->hash = $hash;
-        $reset->created_at = date('Y-m-d H:i:s');
-        $reset->updated_at = date('Y-m-d H:i:s');
-        $this->di['db']->store($reset);
+        // If no recent reset request exists, create a new one
+        if (!$reset instanceof \Model_ClientPasswordReset) {
+            $hash = hash('sha256', time() . random_bytes(13));
+            $reset = $this->di['db']->dispense('ClientPasswordReset');
+            $reset->client_id = $c->id;
+            $reset->ip = $this->ip;
+            $reset->hash = $hash;
+            $reset->created_at = date('Y-m-d H:i:s');
+            $reset->updated_at = date('Y-m-d H:i:s');
+            $this->di['db']->store($reset);
+        }
 
-        // send email
-        $email = [];
-        $email['to_client'] = $c->id;
-        $email['code'] = 'mod_client_password_reset_request';
-        $email['hash'] = $hash;
+        // prepare reset email
+        $email = [
+            'to_client' => $c->id,
+            'code' => 'mod_client_password_reset_request',
+            'hash' => $reset->hash,
+        ];
+        // Send the email if the reset request has the same created_at and updated_at
         $emailService = $this->di['mod_service']('email');
-        $emailService->sendTemplate($email);
+        if ($reset->created_at == $reset->updated_at) {
```

Comment:
If the reset object has just been created, we need to send the email. This can be checked by making sure created at and updated at are the same. 

If we do not send the email in this condition, the email is never sent until the user requests it again - minimum 1 minute later. 

We're also not allowing new password resets to be requested, only to be re-sent. 

You're right, the send_now was missing, I'll fix that :)

Reply:
Okay, that makes sense to me now.
I was thinking about it wrong, thanks!